Automattic/wp-calypso
# migrate external contributors from data getters to ReactQuery

#### Changes proposed in this Pull Request

* Remove data getters related to external contributor status and move to React Query

#### Testing instructions

* Go to `/people/team` on a site with multiple users
* Make sure the checkbox for _This user is a contractor, freelancer, consultant, or agency._ shows the correct value (either ticked if the user is an external contributor or not if they aren't)
* Try to change that value and verify it persists after reload (you should see a network request to `/external-contributors/(add|remove)`
* Attempt to delete the user from the team. In case the user's an external contributor there should be a request to `/remove`. ¹
* Make sure to test error cases by blocking network requests accordingly

¹ In case there's an error ec status removal is still requested. I'm not exactly sure but that's probably wrong and ec status should only be removed if the user is actually removed from the team (i.e. the delete user request is succcessful).
